### ISSUE TITLE
Enhancement: Subtitle additions / overwrite

### ISSUE THREAD

**Issue by AndyZaa** [mjvotaw/plex-sub-downloader#10]
I have plex-sub-downloader installed and working nicely, but it doesn't really address most of my issue with subtitle management in plex_debrid. I get a lot of subtitle tracks of SSA/ASS and PGS which cause Plex to perform some intense transcoding. I can't use the SSA/ASS or PGS subtitles on those videos. SRT subs work fine. 

plex-sub-downloader only works automatically on newly watchlisted vidoes. plex-sub-downloader won't add SRT or any other subs to the Plex database if it sees that there are already any type of existing subs. Deleting the incompatible subs would clear the way to download SRT subs with plex-sub-downloader, at least using the individual metadata key method. However I haven't found any method to manage and delete subs in the Plex database. 

Pasta, https://github.com/cglatot/pasta, comes close but doesn't allow deletion. It would be great if plex-sub-downloader allowed upload of SRT subs to videos that already have subs, or if it provided for deletion of existing subs while uploading SRT subs. As well, if plex-sub-downloader allowed updating videos already in the database (not just new) it would be a total solution.

**Comment by ghost** [comment]
Yeah, I can definitely see the utility of needing to get different subtitle types. I'll have to think on this one a bit. I'm not crazy about the idea of deleting existing subtitles (I mostly just don't trust this not to accidentally break stuff in weird ways), but I absolutely see why you'd want to in a case like this.

And as for updating subs on existing media, I'm hoping to get there soon. I want to get a better user interface/config management put together first though.

**Comment by AndyZaa** [comment]
Yep, I see your point on not deleting existing subs. If plex-sub-downloader allowed upload of subs when there are already existing subs, that would work fine. We could just select the desired sub when playing the video. Thanks for a great utility!

**Comment by ghost** [comment]
You're quite welcome! I'm glad other people are getting some use out of this.

**Comment by ghost** [comment]
I'll be working on this  a bit this week, so hopefully I can get a solution for you figured out.
Also just want to give you a heads up, I'm transferring this repo to my "real" github account. I'm not sure why I started working on this under an alt account in the first place :shrug:

**Comment by mjvotaw** [comment]
Pushed a new release, 0.2.4, last night. I've added a new config option, `format_priority`, that will let you let your preferred subtitle format. Any formats not in that list will be ignored when checking if a video has existing subtitles.

**Comment by AndyZaa** [comment]
Awesome, thank you so much. Dumb question: how do I update to the new version? I ran "pip3 install plex_sub_downloader" and it still comes back as "plex_sub_downloader version 0.2.2".

**Comment by mjvotaw** [comment]
pip3 install --upgrade plex_sub_downloader

**Comment by mjvotaw** [comment]
Hey just a heads up, I just released an update that fixes an issue that was preventing subtitles from actually getting downloaded.

**Comment by AndyZaa** [comment]
Thank you! Updated and will test.
